### PR TITLE
set indent width to 2, convert js and less to it

### PR DESCRIPTION
**Pull request by silverwind**
This updates indentation width to 2 across the code base:

- *.go and Makefile will continue to use tabs but render at width of 2. This should help when working on deeply nested code like templates
- all other files are set to use space indenation (no change here)

Also fixed various discovered indentation and whitespace related issues.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -28,3 +29,12 @@ indent_size = 4
 
 [Makefile]
 indent_style = tab
+indent_size = 8
```
Comment (silverwind) [inline .editorconfig]: I'm not sure what the general preference on tab width for *.go/Makefile is. 8 seems archaic to me but it is what vim and GitHub use by default. I personally prefer 2 for everything, but I guess not everyone might agree to that.

Comment (lafriks) [reply .editorconfig]: 4 is good compromise but I also prefer 2

Comment (silverwind) [reply .editorconfig]: I will update all of them to 2, let's see if it gets some acceptance. This should especially help with template code where the 8-wide tabs really get in the way when one it already 10+ levels deep in indentation.

**Comment by silverwind** [comment]
Updated to 2-space indent now and converted all js and less files to it. This will also make tabs in golang code render as 2-space in editors supporting editorconfig.

**Comment by silverwind** [comment]
Rebased and updated PR description.

**Comment by silverwind** [comment]
When viewing the diff here, set GitHub to hide whitespace-only changes to get a sane view on the actual changes.

**Comment by silverwind** [comment]
Doesn't look like there's much interest in a churn-only commit, so I'll close this.

**Pull request closed without merging**